src/m365/spo/commands/listitem/listitem-roleinheritance-break.ts

```diff
@@ -0,0 +1,107 @@
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { validation } from '../../../../utils';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  id: number;
+  listId?: string;
+  listTitle?: string;
+  clearExistingPermissions?: boolean;
+}
+
+class SpoListItemRoleInHeritanceBreakCommand extends SpoCommand {
+  public get name(): string {
+    return commands.LISTITEM_ROLEINHERITANCE_BREAK;
+  }
+
+  public get description(): string {
+    return 'Break inheritance of list item. Keeping existing permissions is the default behavior.';
+  }
+
+  public optionSets(): string[][] | undefined {
+    return [
+      [ 'listId', 'listTitle' ]
+    ];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (this.verbose) {
+      logger.logToStderr(`Breaking role inheritance of list item in site at ${args.options.webUrl}...`);
+    }
+
+    let requestUrl: string = `${args.options.webUrl}/_api/web/lists`;
+
+    if (args.options.listId) {
+      requestUrl += `(guid'${encodeURIComponent(args.options.listId)}')`;
+    }
+    else {
+      requestUrl += `/getbytitle('${encodeURIComponent(args.options.listTitle as string)}')`;
+    }
+
+    let keepExistingPermissions: boolean = true;
+    if (args.options.clearExistingPermissions) {
+      keepExistingPermissions = !args.options.clearExistingPermissions;
+    }
+
+    const requestOptions: any = {
+      url: `${requestUrl}/items(${args.options.id})/breakroleinheritance(${keepExistingPermissions})`,
+      method: 'POST',
+      headers: {
+        'accept': 'application/json;odata=nometadata',
+        'content-type': 'application/json'
+      },
+      responseType: 'json'
+    };
+
+    request
+      .post(requestOptions)
+      .then(_ => cb(), (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '-i, --id <id>'
+      },
+      {
+        option: '-l, --listId [listId]'
+      },
+      {
+        option: '-t, --listTitle [listTitle]'
+      },
+      {
+        option: '-c, --clearExistingPermissions'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
```

Comment:
Let's extend validation with checking that the value specified in the `id` option is indeed a number.

Reply:
✅